File: tests/test_analyzer.go
package main

import (
    "testing"
)

func TestWordCount(t *testing.T) {
    text := "This is a test."
    result := analyzeText(text)
    if result["word_count"] != 5 {
        t.Errorf("Expected 5, got %v", result["word_count"])
    }
}

func TestSentenceCount(t *testing.T) {
    text := "This is a test. Here's another sentence."
    result := analyzeText(text)
    if result["sentence_count"] != 2 {
        t.Errorf("Expected 2, got %v", result["sentence_count"])
    }
}

func TestCharacterCount(t *testing.T) {
    text := "Hello!"
    result := analyzeText(text)
    if result["character_count"] != 6 {
        t.Errorf("Expected 6, got %v", result["character_count"])
    }
}

func TestMostFrequentWord(t *testing.T) {
    text := "apple apple banana"
    result := analyzeText(text)
    if result["word_frequency"]["apple"] != 2 {
        t.Errorf("Expected 2 for 'apple', got %v", result["word_frequency"]["apple"])
    }
}

File: tests/test_analyzer.py
import unittest
from src.analyzer import analyze_text

class TestTextAnalyzer(unittest.TestCase):

    def test_word_count(self):
        text = "This is a test."
        result = analyze_text(text)
        self.assertEqual(result['word_count'], 5)

    def test_sentence_count(self):
        text = "This is a test. Here's another sentence."
        result = analyze_text(text)
        self.assertEqual(result['sentence_count'], 2)

    def test_character_count(self):
        text = "Hello!"
        result = analyze_text(text)
        self.assertEqual(result['character_count'], 6)

    def test_most_frequent_word(self):
        text = "apple apple banana"
        result = analyze_text(text)
        self.assertEqual(result['word_frequency']['apple'], 2)

if __name__ == '__main__':
    unittest.main()
